drivers/scheduler/k8s/k8s.go

```diff
@@ -429,10 +439,21 @@ func (k *k8s) createNamespace(app *spec.AppSpec, namespace string) (*v1.Namespac
 	return nsObj.(*v1.Namespace), nil
 }
 
-func (k *k8s) createStorageObject(spec interface{}, ns *v1.Namespace, app *spec.AppSpec) (interface{}, error) {
+func (k *k8s) createStorageObject(spec interface{}, ns *v1.Namespace, app *spec.AppSpec, storageprovisioner string) (interface{}, error) {
 	k8sOps := k8s_ops.Instance()
+
 	if obj, ok := spec.(*storage_api.StorageClass); ok {
+		logrus.Infof("[%v] Requested provisioner: %s", app.Key, storageprovisioner)
 		obj.Namespace = ns.Name
+		if storageProvisioner, ok := provisioners[storageprovisioner]; ok {
+			obj.Provisioner = storageProvisioner
+		} else {
+			return nil, &scheduler.ErrFailedToGetStorage{
+				App:   app,
+				Cause: fmt.Sprintf("Failed to get storage: %v. Err: %v", storageprovisioner, fmt.Errorf("Invalid storage provider")),
```

Comment:
you don't need to fmt.Errorf here since you are just printing the string and not returning an err.

Reply:
Fixed.